zerver/lib/subdomains.py

```diff
@@ -24,7 +24,8 @@ def get_subdomain(request):
 def is_subdomain_root_or_alias(request):
     # type: (HttpRequest) -> bool
     subdomain = _extract_subdomain(request)
-    return not subdomain or subdomain in settings.ROOT_SUBDOMAIN_ALIASES
+    return (subdomain == Realm.SUBDOMAIN_FOR_ROOT_DOMAIN
+            or subdomain in settings.ROOT_SUBDOMAIN_ALIASES)
```

Comment:
This particular call point change is arguably incorrect, since `_extract_subdomain` is parsing code that will return `""` unconditionally for the root domain, not `SUBDOMAIN_FOR_ROOT_DOMAIN` (though of course doesn't matter for now since they're the same).  The latent bug gets fixed a few commits later, so not a big deal.

Reply:
Yeah, I took the interface of `_extract_subdomain` to be that it returns SUBDOMAIN_FOR_ROOT_DOMAIN in the case of the root domain, and its implementation is just another case of the implicit use of `""` for that.

But as you say, it doesn't matter because it gets factored away a few commits later anyway.